common/bucket_utils.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for interacting with the buckets."""
+
+from common import logs
+from common import experiment_utils
+
+logger = logs.Logger('bucket_utils')
+
+
+def __using_gsutil():
+    """ Test whether using gsutil format paths. """
+    try:
+        experiment_path_format = experiment_utils.get_cloud_experiment_path()
+    except KeyError:
+        experiment_path_format = 'gs://default'
+
+    return experiment_path_format.startswith('gs://')
+
+
+if __using_gsutil():
```

Comment:
Is this how we plan to use this long term, or is this a temporary solution? Would it make sense to use class inheritance instead longer term?

Reply:
Classes seem like the more conservative solution but I can't really think of an advantage it provides over using this pattern (assuming this module created an object from the class it wanted instead of importing the module it wants) since there's no state being stored. WDYT?

Reply:
Right, i agree, it probably wouldn't have huge benefits, but might be cleaner/more conventional/easier to read? Some potential smaller benefits that come to mind :
- we wouldn't need to have these wrapper functions in this file that explicitly call the real implementation (e.g., `return base_utils.cp(*cp_arguments, **kwargs)`)
- using an abc (https://docs.python.org/3/library/abc.html) might have further benefits, e.g., throwing an error if some abstract method is not implemented
- testing might be cleaner, maybe?

Reply:
Decent point, I don't love the idea of forcing this pattern though:
```
import bucket_utils

bucket_utils_obj = bucket_utils.get_obj()

def f():
  ...
  bucket_utils_obj.do_something()
  ...
```

I think the wrapper functions are good because they avoid this.

Reply:
i'd probably instantiate the global object in the the utils module itself, not at each use site

anyway, it's fine to leave as is in this PR, but might worth considering switching to inheritance in the following ones

Reply:
Let's use the current way for integration. The convenient way is for the user usages. Users only need to `from common import filestore_utils` and call `filestore_utils.<operation_name>`. They don't need to create a object to call such operation.